src/components/tl/ucp/allgatherv/allgatherv_ring.c

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "config.h"
+#include "tl_ucp.h"
+#include "allgatherv.h"
+#include "core/ucc_progress_queue.h"
+#include "utils/ucc_math.h"
+#include "tl_ucp_sendrecv.h"
+
+ucc_status_t ucc_tl_ucp_allgatherv_ring_progress(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_ucp_task_t *task     = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
+    ucc_tl_ucp_team_t *team     = task->team;
+    ucc_rank_t         grank    = team->rank;
+    ucc_rank_t         gsize    = team->size;
+    ptrdiff_t          rbuf     = (ptrdiff_t)task->args.dst.info_v.buffer;
+    ucc_memory_type_t  rmem     = task->args.dst.info_v.mem_type;
+    ucc_datatype_t     rdt_size = ucc_dt_size(task->args.dst.info_v.datatype);
+    ucc_rank_t         sendto   = (grank + 1) % gsize;
+    ucc_rank_t         recvfrom = (grank - 1 + gsize) % gsize;
+    ucc_rank_t         chunk;
```

Comment:
minor: for readability i would declare 2 vars instead of "chunk": send_idx, recv_idx (your args_get_count/displ functions take "idx" as last arg, not "chunk").